endpointmanager/pkg/chplquerier/criteriaquerier.go

```diff
@@ -0,0 +1,174 @@
+	CertificationEdition   string `json:"certificationEdition"`
+	Description            string `json:"description"`
+	Removed                bool   `json:"removed"`
+}
+
+// GetCHPLCriteria queries CHPL for its certification criteria using 'cli' and stores the criteria in 'store'
+// within the given context 'ctx'.
+func GetCHPLCriteria(ctx context.Context, store *postgresql.Store, cli *http.Client) error {
+	log.Debug("requesting certification criteria from CHPL")
+	critJSON, err := getCriteriaJSON(ctx, cli)
+	if err != nil {
+		return err
+	}
+	log.Debug("done requesting certification criteria from CHPL")
+
+	log.Debug("converting chpl json into certification criteria objects")
+	critList, err := convertCriteriaJSONToObj(ctx, critJSON)
+	if err != nil {
+		return errors.Wrap(err, "converting certification criteria JSON into a 'chplCriteriaList' object failed")
+	}
+	log.Debug("done converting chpl json into certification criteria objects")
+
+	log.Debug("persisting certification criteria")
+	err = persistCriterias(ctx, store, critList)
+	log.Debug("done persisting certification criteria")
+	return errors.Wrap(err, "persisting the list of retrieved certification criteria failed")
+}
+
+// makes the request to CHPL and returns the byte string
+func getCriteriaJSON(ctx context.Context, client *http.Client) ([]byte, error) {
+	chplURL, err := makeCHPLCriteriaURL()
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating CHPL certification criteria URL")
+	}
+
+	// None of the returned errors should break the system, so print a warning instead
+	jsonBody, err := getJSON(ctx, client, chplURL)
+	if err != nil {
+		log.Warnf("Got error:\n%s\n\nfrom URL: %s", err.Error(), chplURL.String())
+	}
+	return jsonBody, nil
+}
+
+// builds the CHPL Certification Criteria URL
+func makeCHPLCriteriaURL() (*url.URL, error) {
+	queryArgs := make(map[string]string)
+
+	chplURL, err := makeCHPLURL(chplAPICertCriteriaPath, queryArgs)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating the URL to query CHPL failed")
+	}
+
+	return chplURL, nil
+}
+
+// takes the json byte string and converts it into the associated JSON model
+func convertCriteriaJSONToObj(ctx context.Context, critJSON []byte) (*chplCertifiedCriteriaList, error) {
+	var critList chplCertifiedCriteriaList
+
+	// don't unmarshal the JSON if the context has ended
+	select {
+	case <-ctx.Done():
+		return nil, errors.Wrap(ctx.Err(), "Unable to convert certified criteria JSON to objects - context ended")
+	default:
+		// ok
+	}
+
+	err := json.Unmarshal(critJSON, &critList)
+	if err != nil {
+		return nil, errors.Wrap(err, "unmarshalling the JSON into a chplCertifiedCriteriaList object failed.")
+	}
+
+	return &critList, nil
+}
+
+// takes the JSON model and converts it into an endpointmanager.CertificationCriteria
+func parseHITCriteria(ctx context.Context, criteria *chplCertCriteria, store *postgresql.Store) (*endpointmanager.CertificationCriteria, error) {
```

Comment:
I think you can remove the ctx and store from the function parameter since it's not being used

Reply:
good point, i've removed it!